expr/functions/groupByNode/function_test.go

```diff
@@ -115,35 +143,62 @@ func TestGroupByNode(t *testing.T) {
 			},
 		},
 		{
-			Target: "groupByNode(metric1.foo.*.*,2,\"sum\")",
+			Name:   "groupByNodes_out_of_range_node_is_ignored",
+			Target: "groupByNodes(metric1.foo.*.*,\"sum\",0,5,2)",
 			M: map[parser.MetricRequest][]*types.MetricData{
 				mr: {
-					types.MakeMetricData("metric1.foo.Ab1==.lag", []float64{1, 2, 3, 4, 5}, 1, now32),
-					types.MakeMetricData("metric1.foo.bC2=.lag", []float64{6, 7, 8, 9, 10}, 1, now32),
-					types.MakeMetricData("metric1.foo.Ab1==.lag", []float64{11, 12, 13, 14, 15}, 1, now32),
-					types.MakeMetricData("metric1.foo.bC2=.lag=", []float64{7, 8, 9, 10, 11}, 1, now32),
+					types.MakeMetricData("metric1.foo.bar1.baz", []float64{1, 2, 3, 4, 5}, 1, now32),
 				},
 			},
-			Name: "groupByNode_names_with_special_symbol_equal",
 			Results: map[string][]*types.MetricData{
-				"Ab1==": {types.MakeMetricData("Ab1==", []float64{12, 14, 16, 18, 20}, 1, now32)},
-				"bC2=":  {types.MakeMetricData("bC2=", []float64{13, 15, 17, 19, 21}, 1, now32)},
+				"metric1.bar1": {types.MakeMetricData("metric1.bar1", []float64{1, 2, 3, 4, 5}, 1, now32)},
 			},
 		},
 		{
-			Target: "groupByNode(metric1.foo.*.*,3,\"sum\")",
+			Name:   "groupByNodes_tags_and_nodes_combined",
+			Target: `groupByNodes(metric1.foo.*.*,"sum","name",1)`,
 			M: map[parser.MetricRequest][]*types.MetricData{
 				mr: {
-					types.MakeMetricData("metric1.foo.Ab1==.lag;tag1=value1", []float64{1, 2, 3, 4, 5}, 1, now32),
-					types.MakeMetricData("metric1.foo.Ab2.lag=;tag1=value1", []float64{1, 0, 3, 4, 5}, 1, now32),
+					types.MakeMetricData("metric1.foo.bar1.baz", []float64{1, 2, 3, 4, 5}, 1, now32),
+					types.MakeMetricData("metric1.foo.bar1.bla", []float64{1, 2, 3, 4, 5}, 1, now32),
 				},
 			},
-			Name: "groupByNode_tagged_names_with_special_symbol_equal",
 			Results: map[string][]*types.MetricData{
-				"lag":  {types.MakeMetricData("lag", []float64{1, 2, 3, 4, 5}, 1, now32)},
-				"lag=": {types.MakeMetricData("lag=", []float64{1, 0, 3, 4, 5}, 1, now32)},
+				"metric1.foo.bar1.baz.foo": {types.MakeMetricData("metric1.foo.bar1.baz.foo", []float64{1, 2, 3, 4, 5}, 1, now32)},
+				"metric1.foo.bar1.bla.foo": {types.MakeMetricData("metric1.foo.bar1.bla.foo", []float64{1, 2, 3, 4, 5}, 1, now32)},
 			},
 		},
+		// FIXME: I don't think `..foo==.bar` is a valid metric name, that is making these tests fail https://github.com/grafana/carbonapi/issues/68
```

Comment:
It's not forbidden symbols. Not used for first tags ident.

Reply:
Solved here [`d77e3ee` (#686)](https://github.com/go-graphite/carbonapi/pull/686/commits/d77e3ee963f7262ee852034f6fdcfa27663bca92)